packages/gestalt/src/Button.js

```diff
@@ -136,159 +106,141 @@ function InternalButtonContent({
-          size={size}
-        />
-      </InternalLink>
+const ButtonWithForwardRef: AbstractComponent<Props, unionRefs> = forwardRef<Props, unionRefs>(
+  function Button(
+    {
+      accessibilityLabel,
+      color = 'gray',
+      dataTestId,
+      disabled = false,
+      fullWidth = false,
+      iconEnd,
+      onClick,
+      tabIndex = 0,
+      selected = false,
+      size = 'md',
+      text,
+      type,
+      ...props
+    }: Props,
+    ref,
+  ): Node {
+    const innerRef = useRef<null | HTMLButtonElement>(null);
+
+    // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
+    // that renders <Button ref={inputRef} /> to call inputRef.current.focus()
+    useImperativeHandle(ref, () => innerRef.current);
+
+    const {
+      compressStyle,
+      isTapping,
+      handleBlur,
+      handleMouseDown,
+      handleMouseUp,
+      handleTouchStart,
+      handleTouchMove,
+      handleTouchCancel,
+      handleTouchEnd,
+    } = useTapFeedback({
+      height: innerRef?.current?.clientHeight,
+      width: innerRef?.current?.clientWidth,
+    });
+
+    const { name: colorSchemeName } = useColorScheme();
+    // We need to make a few exceptions for accessibility reasons in darkMode for red buttons
+    const isDarkMode = colorSchemeName === 'darkMode';
+    const isDarkModeRed = isDarkMode && color === 'red';
+    const isDarkModeBlue = isDarkMode && color === 'blue';
+
+    let colorClass = color === 'transparentWhiteText' ? 'transparent' : color;
+
+    if (isDarkModeRed) {
+      colorClass = 'darkModeRed';
+    }
+
+    const { isFocusVisible } = useFocusVisible();
+
+    const sharedTypeClasses = classnames(styles.button, focusStyles.hideOutline, {
+      [styles.inline]: !fullWidth,
+      [styles.block]: fullWidth,
+      [focusStyles.accessibilityOutline]: !disabled && isFocusVisible,
+    });
+
+    const baseTypeClasses = classnames(sharedTypeClasses, touchableStyles.tapTransition, {
+      [styles.sm]: size === 'sm',
+      [styles.md]: size === 'md',
+      [styles.lg]: size === 'lg',
+      // $FlowFixMe[invalid-computed-prop]
+      [styles[colorClass]]: !disabled && !selected,
+      [styles.selected]: !disabled && selected,
+      [styles.disabled]: disabled,
+      [styles.enabled]: !disabled,
+      [touchableStyles.tapCompress]: !disabled && isTapping,
+    });
+
+    const parentButtonClasses = classnames(sharedTypeClasses, styles.parentButton);
+
+    const childrenDivClasses = classnames(baseTypeClasses, styles.childrenDiv);
+
+    const textColor =
+      (disabled && 'subtle') ||
+      (selected && 'inverse') ||
+      ((isDarkModeRed || isDarkModeBlue) && 'default') ||
+      DEFAULT_TEXT_COLORS[color];
+
+    const buttonText = (
+      <Text align="center" color={textColor} overflow="normal" weight="bold">
+        {text}
+      </Text>
     );
-  }
 
-  if (props.type === 'submit') {
-    const { name } = props;
+    // if (type === 'submit') {
```

Comment:
don't remove this line. just flatten the Flow Types

Reply:
Do you mean I shouldn't merge the button type implementations, just remove the submit button type union?
I reverted the removal of the lines, is it all right now?

Reply:
DONE